poc_iot_verifier/src/loader.rs

```diff
@@ -157,34 +162,57 @@ impl Loader {
                     LoraBeaconIngestReportV1::decode(buf)?.try_into()?;
                 tracing::debug!("beacon report from ingestor: {:?}", &beacon);
                 let packet_data = beacon.report.data.clone();
-                tracing::debug!("beacon data: {:?}", &packet_data);
-                Report::insert_into(
-                    &self.pool,
-                    beacon.ingest_id(),
-                    beacon.report.remote_entropy,
-                    packet_data,
-                    buf.to_vec(),
-                    &beacon.received_timestamp,
-                    ReportType::Beacon,
-                )
-                .await
+                match self.check_valid_gateway(&beacon.report.pub_key).await {
+                    true => {
+                        Report::insert_into(
+                            &self.pool,
+                            beacon.ingest_id(),
+                            beacon.report.remote_entropy,
+                            packet_data,
+                            buf.to_vec(),
+                            &beacon.received_timestamp,
+                            ReportType::Beacon,
+                        )
+                        .await
+                    }
+                    false => {
+                        // if a gateway doesnt exist then drop the report
+                        tracing::warn!(
+                            "dropping beacon report as gateway not found: {:?}",
+                            &beacon
+                        );
+                        Ok(())
+                    }
+                }
             }
             FileType::LoraWitnessIngestReport => {
                 let witness: LoraWitnessIngestReport =
                     LoraWitnessIngestReportV1::decode(buf)?.try_into()?;
                 tracing::debug!("witness report from ingestor: {:?}", &witness);
                 let packet_data = witness.report.data.clone();
-                tracing::debug!("witness data: {:?}", &packet_data);
-                Report::insert_into(
-                    &self.pool,
-                    witness.ingest_id(),
-                    Vec::<u8>::with_capacity(0),
-                    packet_data,
-                    buf.to_vec(),
-                    &witness.received_timestamp,
-                    ReportType::Witness,
-                )
-                .await
+                match self.check_valid_gateway(&witness.report.pub_key).await {
```

Comment:
Similar thing here, could potentially make a function to add both types of reports?